src/Microsoft.Framework.PackageManager/List/LoadableAssemblyWalker.cs

```diff
@@ -0,0 +1,135 @@
+        private readonly string _runtimeFolder;
+
+        public LoadableAssemblyWalker(
+            FrameworkName framework,
+            ApplicationHostContext hostContext,
+            IDictionary<Library, LibraryDescription> libDictionary,
+            string runtimeFolder)
+        {
+            _framework = framework;
+            _hostContext = hostContext;
+            _runtimeFolder = runtimeFolder;
+            _libDictionary = libDictionary;
+
+            _gacResolver = new GacDependencyResolver();
+        }
+
+        public ISet<string> Walk(IGraphNode<Library> root)
+        {
+            var assemblies = new HashSet<string>();
+            var libraries = new HashSet<Library>();
+            root.DepthFirstPreOrderWalk(visitNode: (node, _) => VisitLibrary(node, _, libraries, assemblies));
+
+            return assemblies;
+        }
+
+        private bool VisitLibrary(IGraphNode<Library> node, IEnumerable<IGraphNode<Library>> ancestors, ISet<Library> visitedLibraries, ISet<string> assemblies)
+        {
+            if (visitedLibraries.Add(node.Item))
+            {
+                LibraryDescription desc;
+                if (_libDictionary.TryGetValue(node.Item, out desc))
+                {
+                    foreach (var loadableAssembly in desc.LoadableAssemblies)
+                    {
+                        DepthFirstGraphTraversal.PreOrderWalk(
+                            root: loadableAssembly,
+                            visitNode: (assemblyNode, assemblyAncestors) => VisitAssembly(assemblyNode, assemblyAncestors, assemblies),
+                            getChildren: GetAssemblyDependencies);
+                    }
+                }
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private IEnumerable<string> GetAssemblyDependencies(string assemblyName)
+        {
+            var filepath = ResolveAssemblyFilePath(assemblyName);
+            if (filepath != null && File.Exists(filepath))
+            {
+                var assemblyInfo = new AssemblyInformation(filepath, processorArchitecture: null);
+                return assemblyInfo.GetDependencies();
+            }
+            else
+            {
+                return Enumerable.Empty<string>();
+            }
+        }
+
+        private bool VisitAssembly(string assembly, IEnumerable<string> ancestors, ISet<string> assemblies)
+        {
+            // determine if keep walking down the path
+            if (ancestors.Any(a => a == assembly))
+            {
+                // break the reference loop
+                return false;
+            }
+
+            var filepath = ResolveAssemblyFilePath(assembly);
+            if (filepath == null)
+            {
+                if (_framework != AspNetCore50)
+                {
+                    var gacDesc = _gacResolver.GetDescription(new Library { Name = assembly, IsGacOrFrameworkReference = true }, _framework);
```

Comment:
For now the GAC resolving is not actually needed since we ignore all the unresolved assemblies here. I'll remove the code.